static/panes/compiler.js

```diff
@@ -130,33 +129,77 @@ function Compiler(hub, container, state) {
 
     this.fontScale = new FontScale(this.domRoot, state, this.outputEditor);
 
-    this.compilerPicker.selectize({
+    var self = this;
+    var optgroups = this.compilerService.getGroupsInUse(this.currentLangId);
+    optgroups.unshift({value:'favs',label:'Favorites'});
+
+    this.compilerSelectizer = new TomSelect(this.compilerPicker[0],{
         sortField: this.compilerService.getSelectizerOrder(),
         valueField: 'id',
         labelField: 'name',
         searchField: ['name'],
         optgroupField: 'group',
-        optgroups: this.compilerService.getGroupsInUse(this.currentLangId),
+        optgroups: optgroups,
         lockOptgroupOrder: true,
         options: _.filter(this.getCurrentLangCompilers(), function (e) {
             return !e.hidden || e.id === state.compiler;
         }),
         items: this.compiler ? [this.compiler.id] : [],
         dropdownParent: 'body',
         closeAfterSelect: true,
-    }).on('change', _.bind(function (e) {
-        var val = $(e.target).val();
-        if (val) {
-            ga.proxy('send', {
-                hitType: 'event',
-                eventCategory: 'SelectCompiler',
-                eventAction: val,
-            });
-            this.onCompilerChange(val);
+        plugins:['dropdown_input'],
+        onChange: function (val){
+            if (val) {
+                ga.proxy('send', {
+                    hitType: 'event',
+                    eventCategory: 'SelectCompiler',
+                    eventAction: val,
+                });
+                self.onCompilerChange(val);
+            }
+        },
+        duplicates: true,
```

Comment:
Do we have duplicates anywhere?

Reply:
All of the "Favorites" are duplicates. If a user stars "armv7-a clang 11.0.1" it will be displayed in two optgroups: Favorites and ARM 32-bit clang

Reply:
Got it! thanks